src/plugins/WebOfTrust/util/TickerDelayedBackgroundJob.java

```diff
@@ -0,0 +1,363 @@
+                return;
+            case WAITING:
+                toTERMINATED();
+                return;
+            case RUNNING:
+                toTERMINATING();
+                return;
+        }
+    }
+
+    @Override
+    public synchronized boolean isTerminated() {
+        return state == JobState.TERMINATED;
+    }
+
+    @Override
+    public synchronized void waitForTermination(long timeout) throws InterruptedException {
+        long deadline = System.currentTimeMillis() + timeout;
+        while(timeout > 0 && state != JobState.TERMINATED) {
+            wait(timeout);
+            timeout = deadline - System.currentTimeMillis();
+        }
+    }
+
+    /**
+     * Implementation of {@link #triggerExecution(long)}.
+     * Caller must ensure synchronization on {@code this}.
+     */
+    private void tryEnqueue(long delay) {
+        if (state == JobState.TERMINATING || state == JobState.TERMINATED) {
+            return;
+        }
+        if (delay < 0) {
+            delay = 0;
+        }
+        long newExecutionTime = System.currentTimeMillis() + delay;
+        if (newExecutionTime < nextExecutionTime) {
+            nextExecutionTime = newExecutionTime;
+            if (state == JobState.RUNNING) {
+                // Will automatically schedule this run when the running job finishes.
+                return;
+            }
+            if (state == JobState.WAITING) {
+                // Best-effort attempt at removing the stale job to be replaced.
+                ticker.removeQueuedJob(waitingTickerJob);
+                // Replace the ticker job in case the above fails, so the stale job will not run.
+                waitingTickerJob = createTickerJob();
+            }
+            enqueueWaitingTickerJob(delay);
+        }
+    }
+
+    /**
+     * Schedules the ticker job (which will submit the background job to the executor) at the
+     * ticker with the given delay, or executes the ticker job immediately if the delay is zero.
+     * If the current state is {@code IDLE}, we the state is changed to {@code WAITING} and a new
+     * ticker job is created.
+     * Caller must ensure synchronization on {@code this}.
+     * @param delay the delay in ms
+     */
+    private void enqueueWaitingTickerJob(long delay) {
+        assert(state == JobState.IDLE || state == JobState.WAITING) :
+                "enqueueing ticker job in non-IDLE and non-WAITING state";
+        // Use a unique job for each (re)scheduling to avoid running twice.
+        if (state == JobState.IDLE) {
+            toWAITING();
+        }
+        if (delay > 0) {
+            ticker.queueTimedJob(waitingTickerJob, name + " (waiting)", delay, true, false);
+        } else {
+            waitingTickerJob.run();
+        }
+    }
+
+    /**
+     * Creates a tiny job that instructs the executor to run the background job immediately, to
+     * be executed by the ticker.
+     */
+    private Runnable createTickerJob() {
```

Comment:
Not a blocker as it is quite a bit of refactoring, just providing this in case you're still interested in doing much polishing after all my comments :)

While reviewing the class, I noticed that it took quite a bit of time for my brain to memorize the different threads / runnables involved in the whole class. There are 4 of those:
(1) The actual Runnable from the user which is passed to the constructor
(2) The anonymous FastRunnable returned here by createTickerJob() whose purpose it is to avoid duplicates
(3) The DelayedBackgroundRunnable which is run by the FastRunnable (2)
(4) The actual thread which (3) runs on, which will be stored in the member variable "thread". This is not a Runnable, but in my experience the brain stores this in the same area as a Runnable, so it grows the list of things to not mix up.

You could maybe cut down on the amount of things to confuse by merging the Runnable (2) returned by this function here with the Runnable number (3). That would mean that it couldn't be recycled for multiple executions, but this class is supposed to deal with jobs which occur in the dimension of > 1 millisecond (resolution is milliseconds), so object construction won't be very frequent anyway.